jupyter_server/services/contents/handlers.py

```diff
@@ -116,18 +117,26 @@ async def get(self, path=""):
         content = int(content_str or "")
 
         if not cm.allow_hidden and await ensure_async(cm.is_hidden(path)):
-            raise web.HTTPError(404, f"file or directory {path!r} does not exist")
-
-        model = await ensure_async(
-            self.contents_manager.get(
-                path=path,
-                type=type,
-                format=format,
-                content=content,
+            self.set_status(HTTPStatus.NOT_FOUND)
+            self.write(f"file or directory {path!r} does not exist")
+            await self.finish()
```

Comment:
It seems like it could be extracted into a function for *clean error response* like `_finish_model`: https://github.com/bloomsa/jupyter_server/blob/78394daec886b9b323b4801380569bd8ef966fc9/jupyter_server/services/contents/handlers.py#L86

I haven't figured out how to name this function yet though, FYI 😄

```python
async def _finish_error(self, code, msg):
  self.set_status(code)
  self.write(msg)
  await self.finish()
```

Reply:
good suggestion 👍